runner/runner.cc

```diff
@@ -299,17 +303,15 @@ main(int argc, char** argv)
                 // Run map_state_preparation program via bpf_prog_test_run_opts.
                 std::vector<uint8_t> data_in(1024);
                 std::vector<uint8_t> data_out(1024);
-                std::vector<uint8_t> context_in(4);
-                std::vector<uint8_t> context_out(4);
 
                 bpf_test_run_opts opts;
                 memset(&opts, 0, sizeof(opts));
                 opts.sz = sizeof(opts);
                 opts.repeat = prep_program_iterations;
-                opts.data_in = data_in.data();
-                opts.data_out = data_out.data();
-                opts.data_size_in = static_cast<uint32_t>(data_in.size());
-                opts.data_size_out = static_cast<uint32_t>(data_out.size());
+                opts.ctx_in = data_in.data();
```

Comment:
Do we need to populate `ctx_in` even in case of XDP program type? I think this file uses `BPF_PROG_TYPE_XDP` for Linux and `BPF_PROG_TYPE_SOCK_OPS` for Windows.

Reply:
Good point. In my understanding, data_in and data_out are sufficient for XDP programs. data_in is mandatory.

Reply:
It depends on the specific attach point.

XDP -> data mandatory, context optional.
sockops -> data prohibited, context required.